signal_scanner_bot/env.py

```diff
@@ -1,20 +1,56 @@
 import logging
 import os
+from pathlib import Path
 from threading import Lock
-from typing import Any, Callable, List, Set
+from typing import Any, Optional, Callable, List, Set
 
 
 log = logging.getLogger(__name__)
 
 
 _VARS = []
+START_LISTENING = "AUTOSCANON"
+STOP_LISTENING = "AUTOSCANOFF"
+START_LISTENING_NOTIFICATION = "==Auto Scanning Activated=="
+STOP_LISTENING_NOTIFICATION = "==Auto Scanning Deactivated=="
 
 
 class _State:
     """Class for holding global state across threads/tasks"""
 
-    LISTENING = False
-    STOP_REQUESTED = False
+    # Initialize state, set LISTENING to true if state file exists,
+    # sets false if not.
+    def __init__(self, file: Path):
+        self.file = file
+        self.LISTENING = self.file.exists()
+        self.STOP_REQUESTED = False
+
+    # Method to update the listening status of the State class
+    # object. Checks for on/off and creates/deletes state file.
+    def update_listening_status(self, status: str) -> str:
+        if status == START_LISTENING:
+            self.LISTENING = True
+            self.file.touch()
+            return START_LISTENING_NOTIFICATION
+        elif status == STOP_LISTENING:
+            self.LISTENING = False
+            self.file.unlink(missing_ok=True)
+            return STOP_LISTENING_NOTIFICATION
+        else:
+            raise ValueError(
+                f"Value {status} provided to _State object, expected AUTOSCANON or AUTOSCANOFF."
```

Comment:
Very minor change for readability
```suggestion
                f"Value '{status}' provided to _State object, expected {START_LISTENING} or {STOP_LISTENING}."
```